src/XMLimport.cpp

```diff
@@ -149,7 +152,37 @@ bool XMLimport::importPackage(QIODevice* device, QString packName, int moduleFla
         readNext();
 
         if (isStartElement()) {
-            if (name() == "MudletPackage") {
+            if (name() == QStringLiteral("MudletPackage")) {
+                QString versionString;
+                if (attributes().hasAttribute(QStringLiteral("version"))) {
+                    versionString = attributes().value(QStringLiteral("version")).toString();
+                    if (!versionString.isEmpty()) {
+                        bool isOk = false;
+                        float versionNumber = versionString.toFloat(&isOk);
+                        if (isOk) {
+                            mVersionMajor = qFloor(versionNumber);
+                            mVersionMinor = qRound(1000.0 * versionNumber) - (1000 * mVersionMajor);
+                        }
+                        if (pVersionString) {
+                            *pVersionString = versionString;
+                        }
+                    }
+                }
+
+                if (mVersionMajor > 1
+                    /*||(mVersionMajor==1&&mVersionMinor)*/) {
+                    // Minor check is not currently relevant, just abort on 2.000f or more
+
+                    QString moanMsg = tr("[ ALERT ] - Sorry, the file being read is marked as being version %1 which is\n"
```

Comment:
Yeah, but I retain the ability to make decimal changes to the number so that code can consider extra sub-elements/attributes that older versions will qDebug() message about/ignore respectively - if needed.  Off the top of my head I do not have any thing in mind that I cannot do without just using a pair of elements with an attribute that older code will ignore the first and use the second but newer code can spot.  E.g, storing both a relative file name and an absolute file name for the same file e.g. the sound file for a trigger:

    <mSoundFile isRelative="true">../../sounds/tada.wav</mSoundFile>
    <mSoundFile isRelative="false">/home/stephen/.config/mudlet/sounds/tada.wav</mSoundFile>

Old code will read both entries but the second will override the first, new code will spot the relative attribute and know to use that one - which is portable between different OS platforms because it is relative to the profile's directory.  This I can use without changing a version value.

OTOH I have plans to add (upto) 4 icons to buttons that do not disappear when disabled (due to a new attribute "isNotHiddenWhenDisabled") - I hope and _think_ I can squeeze it into the existing layout in a way that degrades in a similar manner, but in the worse case scenario I _might_ (I am only saying might here) need to flag the code to treat the contents in that element in a different way - thinking harder I can't come up with a concrete use case, but it will remain a tool available - should the need arise - to nudge that value up by 0.001...